From: David Milburn <dmilburn@redhat.com>
Date: Wed, 18 Aug 2010 23:59:46 -0400
Subject: [ata] sata_mv: Properly initialize main irq mask
Message-id: <1282175987-21696-9-git-send-email-dmilburn@redhat.com>
Patchwork-id: 27697
O-Subject: [RHEL5.6 PATCH 8/9] BZ 554872 sata_mv: Properly initialize main irq
	mask
Bugzilla: 554872
RH-Acked-by: Jeff Garzik <jgarzik@redhat.com>

commit 5d0fb2e730e2085021cf5c8b6d14983e92aea75b
Author: Thomas Reitmayr <treitmayr@devbase.at>
Date:   Sat Jan 24 20:24:58 2009 +0100

    sata_mv: Properly initialize main irq mask

BZ 554872

diff --git a/drivers/ata/sata_mv.c b/drivers/ata/sata_mv.c
index ac6674e..b096885 100644
--- a/drivers/ata/sata_mv.c
+++ b/drivers/ata/sata_mv.c
@@ -3059,6 +3059,9 @@ static int mv_init_host(struct ata_host *host, unsigned int board_idx)
 		hpriv->main_irq_mask_addr  = mmio + PCI_HC_MAIN_IRQ_MASK_OFS;
 	}
 
+	/* initialize shadow irq mask with register's value */
+	hpriv->main_irq_mask = readl(hpriv->main_irq_mask_addr);
+
 	/* global interrupt mask: 0 == mask everything */
 	mv_set_main_irq_mask(host, ~0, 0);
 
